src/librustc_middle/mir/mod.rs

```diff
@@ -72,15 +72,21 @@ impl<'tcx> HasLocalDecls<'tcx> for Body<'tcx> {
 
 /// The various "big phases" that MIR goes through.
 ///
+/// These phases all describe dialects of MIR. Since all MIR uses the same datastructures, the
+/// dialects forbid certain variants or values in certain phases.
+///
 /// Warning: ordering of variants is significant.
 #[derive(Copy, Clone, RustcEncodable, RustcDecodable, Debug, PartialEq, Eq, PartialOrd, Ord)]
 #[derive(HashStable)]
 pub enum MirPhase {
     Build = 0,
     Const = 1,
     Validated = 2,
```

Comment:
Given that we have a MIR validation that has nothing to do with `MirPhase::Validated`, this name is also rather confusing (and pre-existing).

Reply:
I renamed it to `ConstPromotion`, which, after looking at what the query does more closely seems to be the only real thing it does.

Reply:
Ah that makes much more sense. :)
What does `Const` do?

Reply:
No idea, I think that used to be what we used for mir based const eval before miri

Reply:
Can you add a "FIXME: figure out what this phase does" or so?